src/assembler.jl

```diff
@@ -51,3 +51,14 @@ assembled values.
 function end_assemble(a::Assembler)
     return sparse(a.I, a.J, a.V)
 end
+
+"""
+    assemble!(g, ge, edof)
+
+Assembles the element residual `ge` into the global residual vector `g`.
+"""
+@inline function assemble!{T}(g::AbstractVector{T}, ge::AbstractVector{T}, edof::AbstractVector{Int})
+    @inbounds for i in 1:length(ge)
```

Comment:
asserts are typically used for purely internal errors (like getting into an "impossible" state), while throwing is done for user errors.